BB-1205: Cherry pick features/commits from Cloudera

giovannicimolin opened the pull request:
This PR cherry picks a few customizations made to Cloudera's branch into our common branches.

### Changes:
1. Added the [Dashboard announcements feature](https://github.com/edx/edx-platform/pull/16496). This was cherry picked from the upstream PR.
2. Mobile auth completion fix cherry picked from [upstream PR](https://github.com/edx/edx-platform/pull/20369) to `ironwood.master` branch. Should I squash these?
3. [Change for get okta workflow working](https://github.com/open-craft/edx-platform/commit/06153e093a8875e937fd7a2b434012284728fc53) cherry picked, but gated behind waffle flag because I won't be able to test all corner cases for this feature.
4. [Add api support to let users opt out of email updates](https://github.com/open-craft/edx-platform/pull/168/commits/e47fe4ee1f1b2ca5458cda0de606c055098d1d37) cherry-picked and squashed (upstream PR not yet merged).

### Notes
When upgrading the client, we need to make the following changes to the instance:
* Switch on `third_party_auth.enable_okta_auth_fix` waffle switch from the Django Admin.
* See https://github.com/open-craft/configuration/pull/96 for more instance configuration changes.

### Reviewer:
- [x] @pomegranited

pomegranited commented:
@giovannicimolin Thanks for making these changes!  I'll test them on the sandbox when it's running. Few changes requested though:

> * Added the [Dashboard announcements feature](https://github.com/edx/edx-platform/pull/16496). This was cherry picked from the upstream PR.

cf [da91c6d](https://github.com/open-craft/edx-platform/pull/168/commits/da91c6d7f1d47255afcbe8298276f0ab4d3809fd) When you do a cherry-pick, could you include the `-x` command-line option, so it links back to the original commit?  If it's a pain to update this commit to do it here, then don't worry about it now, just remember for future.

> Mobile auth completion fix cherry picked from [upstream PR](https://github.com/edx/edx-platform/pull/20369) to `ironwood.master` branch. Should I squash these?

I'm really hoping [they'll let me just use the first commit from this PR](https://github.com/edx/edx-platform/pull/20369#issuecomment-491657833) instead of all the other unrelated updates that stemmed from running `make upgrade`.

So yep, go ahead and just use `git cherry-pick -x 5d7d69f3e25ef781ee216ed61c7f19088c1c5726` to add [just the first commit](https://github.com/edx/edx-platform/pull/20369/commits/5d7d69f3e25ef781ee216ed61c7f19088c1c5726) from that PR.

> [Change for get okta workflow working](https://github.com/open-craft/edx-platform/commit/06153e093a8875e937fd7a2b434012284728fc53) cherry picked, but gated behind waffle flag because I won't be able to test all corner cases for this feature.

This change looks good to me!  I'll test it as best I can on the sandbox, and will use this for my upstream PR (to do as part of a separate support ticket).

giovannicimolin commented:
@pomegranited Remade this branch and cherry picked everything with `git cherry-pick -x`. :grin: 
Added that missing PR, squashed the commits and referenced the upstream PR, but preserved the commit messages.
